slothy/targets/arm_v7m/arch_v7m.py

```diff
@@ -1486,7 +1486,7 @@ def make(cls, src):
         obj.increment = None
         obj.pre_index = 0
         obj.addr = obj.args_in[0]
-        obj.args_in_out_different = [(0,0)] # Can't have Rd==Ra
+        #obj.args_in_out_different = [(0,0)] # Can't have Rd==Ra
```

Comment:
Please remove those, not comment them out.
Also we need to test if this affects any other examples in SLOTHY. 

For that please make sure you have a clean copy of SLOTHY, and then run
```
python3 example.py --timeout 60 --only-target=slothy.targets.arm_v7m.cortex_m7
```

This is going to run for a few hours. Then zip up the output files in `examples/opt/armv7m` and attach them to this PR.